.github/docker-images/amazonlinux/Dockerfile

```diff
@@ -31,9 +31,9 @@ RUN curl -sSL https://github.com/Kitware/CMake/releases/download/v3.10.0/cmake-3
 # Install OpenSSL 1.1.1
```

Comment:
@BolongZhang-AWS Can you build each of the 3 container images locally and push to `docker.pkg.github.com/awslabs/aws-iot-device-client/{ubi8|ubuntu-16-x64|amazonlinux}` so that the CI builds are run inside of the containers containing the new dependencies?  I am sending you a dm with the details for how to publish.

Reply:
Addressed this comment with additional commits on this PR.